### PR TITLE
demo/dark-themed-stock-chart

### PR DESCRIPTION
**Pull request by bm64**
Added dark themed stock chart demo.

### CODE REVIEW

**Review by pawellysy — APPROVED** [review]
LGTM!

**Review by TorsteinHonsi — COMMENTED** [review]
Thanks! Did you consider separating out the theming into a `setOptions` call? Otherwise it will be cumbersome for users to modify this for their own charts, since the theming and the general chart setup is intermixed.

**Review by DJTechnoo — APPROVED** [review]
Looking nice!

**Review by kamil-musialowski — APPROVED** [review]
Great!

File: samples/stock/demo/stock-design-dark-theme/demo.details [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+name: Dark themed Stock chart
+authors:
+  - Mateusz Bernacik
+js_wrap: b
+tags:
+  - Highcharts Stock demo
+categories:
+  - General:
```
Comment (pawelfus) [inline samples/stock/demo/stock-design-dark-theme/demo.details:9]: Change category

**Review by pawelfus — CHANGES_REQUESTED** [review]
Just one minor thing

**Review by pawelfus — APPROVED** [review]
Thanks!

File: samples/stock/demo/responsive/scrollbar-disabled/demo.css
Comment (TorsteinHonsi) [inline samples/stock/demo/responsive/scrollbar-disabled/demo.css:1]: This demo is one level too deep in the file structure... Was it intended to be part of the demo pages?

File: samples/stock/demo/stock-design-dark-theme/demo.css
Comment (TorsteinHonsi) [inline samples/stock/demo/stock-design-dark-theme/demo.css:1]: Do we need to repeat the word "stock" in the file path `stock-design-dark-theme`? All demos are stock...

File: samples/stock/demo/responsive/scrollbar-disabled/demo.css
Comment (bm64) [reply samples/stock/demo/responsive/scrollbar-disabled/demo.css:1]: Must have moved it by mistake, sorry!

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks! There are still some elements that differ from the design sketch we are targeting at. We should comply with the sketch, to demonstrate how flexible the Highcharts design is.

* The range selector buttons. The sketch has a simple dropdown. If we choose to use buttons, they must follow the dark theme.
* The scrollbar. The sketch doesn't have one.
* The range selector. The navigator handles look different. 
* The border color of the candles
* The sketch uses a candlestick series in the navigator too. Don't know if that's possible in Highcharts, but I think so.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks! Still some details I think we should work on to make it match the sketch
* Drop the 1px borders on the range selector buttons and the context button. The sketch has borderless icons at the top, it looks cleaner.
* Drop (or dim) the vertical grid lines inside the navigator
* Drop the scrollbar. It doesn't really fit in.

**Review by marvin19 — CHANGES_REQUESTED** [review]
- Remove white border from candlestick (replace with red/green border for red/green points)
- Add `lang.accessibility.defaultChartTitle`
- Add `accessibility.typeDescription`: ‘Combination chart with 2 data series. ‘
- Add title to yAxis
- linkedDescription

For title and description, use original text from dataset if it exists in some other demo. Also when RangeSelector is a dropdown, there seems to be a keyboard navigation issue. Will report it as separate issue.

**Comment by bm64** [comment]
@marvin19 Thanks! I've made the requested changes. Regarding the candlestick border, chaning it would also change the colors of the high and low shapes since it's currently impossible to change the color of the candle separately. In that case is it okay to leave it as is?

**Review by marvin19 — APPROVED** [review]
Keep as it is!

**Review by TorsteinHonsi — COMMENTED** [review]
Thanks! One final question before approving - the demo currently opens with the navigator zoomed all out. In the inspiration sketch, only the right part of the navigator is pre-selected (through `rangeSelector.selected`. Should we extend the example data so that we get to show off more of the navigator's value? Right now it shows a series that exactly reflects the main series, plus the two handles at the end, that kind of just hang there. If we set `rangeSelector.selected` to 3 initially, we get the result I want, but I'm not sure if we want 0.5 hours granularity initially. Maybe it's better to use a longer time series.

**Comment by pawelfus** [comment]
Good point. To me, with our dataset 5min looks the best. Good visible-available data ratio, candles are not too crowded, not too few and far between, right in the spot.

**Comment by bm64** [comment]
@TorsteinHonsi I decided to change the initial timeframe to 5m, as this is an interval frequently used by traders and people who analyze stock charts. The navigator looks decent in this configuration. Let me know what you think 😃

**Review by TorsteinHonsi — APPROVED** [review]
I think that looks very good!

**Review by TorsteinHonsi — APPROVED** [review]
I think that looks very good!

**Pull request merged**